### PR TITLE
Add unit tests for Dispose* functions in backend-shared

### PR DESCRIPTION
**Pull request by Rizwana777**
#### Description:
Moved duplicated lines in tests to BeforeEach block

#### Link to JIRA Story (if applicable):
https://issues.redhat.com/browse/GITOPSRVCE-687

### CODE REVIEW

**Review by chetan-rns — COMMENTED** [review]
Looks good @Rizwana777. I suggest the following changes and I think it  applies to all the files:

1. We can skip the defer in `AfterEach` since there is only one statement in it.
2. Can we verify if the error returned when we try to get a resource after calling Dispose is for no rows found? We can probably use this existing utility function IsResultNotFoundError()

File: backend-shared/db/gitopsengineinstance.go [hunk 1]
```diff
@@ -226,7 +226,7 @@ func (dbq *PostgreSQLDatabaseQueries) internalDeleteGitopsEngineInstanceById(ctx
 	deleteResult, err := dbq.dbConnection.Model(result).WherePK().Context(ctx).Delete()
 	if err != nil {
 		pgErr := err.(pg.Error)
-		return 0, fmt.Errorf("error on deleting operation: %v\nPGError:%v\n", err, pgErr)
+		return 0, fmt.Errorf("error on deleting operation: %v\nPGError:%v\n ", err, pgErr)
```
[the remark this answers is not in the document]
Comment (Rizwana777) [reply backend-shared/db/gitopsengineinstance.go]: This line is showing warning as `error strings should not end with punctuation or newlines (ST1005)` so I have added space at the end, now I removed \n at the end, please let me know if it is required

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, thanks @Rizwana777! One final request: 

It looks like the inside of this block is not covered in DisposeResources/DisposeAppScopedResources:
```go
		disposeErr := resource.Dispose(ctx, dbq)
		if disposeErr != nil {
			if err == nil {
				err = disposeErr
			} else {
				// append the error to the existing error
				err = fmt.Errorf("error: %v. error: %v", disposeErr, err)
			}
		}
```

It should be possible to cover these by creating a mock resource that implements the DisposableResource interface, but have the 'Dispose' function of that interface return an error.

**Comment by jgwest** [comment]
fail on must-gather

/retest

**Pull request merged**